common/models/src/mutable_batch.rs

```diff
@@ -29,12 +29,13 @@ impl MutableBatch {
 
     pub fn column_mut(&mut self, name: &str, col_type: PhysicalCType) -> Result<&mut Column> {
         let columns_len = self.columns.len();
-        let column_idx = self
-            .column_names
-            .raw_entry_mut()
-            .from_key(name)
-            .or_insert_with(|| (name.to_string(), columns_len))
-            .1;
+        if !self.column_names.contains_key(name) {
+            self.column_names.insert(name.to_string(), columns_len);
+        }
+        let column_idx = self.column_names.get_mut(name).ok_or(Error::Common {
```

Comment:
I think it doesn't need to calculate hash for three times, here is an example of simpler code.
 
```rust
let column_idx = match self.column_names.get(name) {
    Some(column_idx) => *column_idx,
    None => {
        let column_idx = self.columns.len();
        self.column_names.insert(name.to_string(), column_idx);
        self.columns.push(Column::new(self.row_count, col_type)?);
        column_idx
    }
};

Ok(&mut self.columns[column_idx] as _)
```